.github/workflows/e2e.yml

```diff
@@ -0,0 +1,87 @@
+name: e2e
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 4 * * *'
+  pull_request:
+    branches:
+      - main
+      - 'release-*'
+  push:
+    branches:
+      - main
+      - 'release-*'
+
+permissions:
+  contents: read
+
+jobs:
```

Comment:
@okartau pointed me to `needs`. To also cover #882 can we have both `images` (from `ci.yaml`) and `e2e` in the same workflow where `e2e` depend on `images`.

Reply:
Than would probably make sense, but not in this PR. I'd prefer to address #822 separately.